src/main/java/com/flightstats/hub/dao/aws/DynamoChannelConfigDao.java

```diff
@@ -84,112 +79,89 @@ public void upsert(ChannelConfig config) {
                 .withKey(keyMap);
         try {
             GetItemResult result = dbClient.getItem(getItemRequest);
             if (result.getItem() == null) {
                 return null;
             }
-            return mapItem(result.getItem());
+            return mapItem(result.getItem())
+                    .orElseThrow(() -> new ResourceNotFoundException("Unable to read channel config from dynamo " + name));
         } catch (ResourceNotFoundException e) {
             log.info("channel not found " + e.getMessage());
             return null;
         }
     }
 
-    private ChannelConfig mapItem(Map<String, AttributeValue> item) {
-        ChannelConfig.ChannelConfigBuilder builder = ChannelConfig.builder()
-                .creationDate(new Date(Long.parseLong(item.get("date").getN())))
-                .name(item.get("key").getS())
-                .displayName(item.get("displayName").getS());
-        if (item.get("ttlDays") != null) {
-            builder.ttlDays(Long.parseLong(item.get("ttlDays").getN()));
-        }
-        if (item.get("keepForever") != null) {
-            builder.keepForever(item.get("keepForever").getBOOL());
-        }
-        if (item.containsKey("description")) {
-            builder.description(item.get("description").getS());
-        }
-        if (item.containsKey("tags")) {
-            builder.tags(item.get("tags").getSS());
-        }
-        if (item.containsKey("replicationSource")) {
-            builder.replicationSource(item.get("replicationSource").getS());
-        }
-        if (item.get("maxItems") != null) {
-            builder.maxItems(Long.parseLong(item.get("maxItems").getN()));
-        }
-        if (item.containsKey("owner")) {
-            builder.owner(item.get("owner").getS());
-        }
-        if (item.containsKey("storage")) {
-            builder.storage(item.get("storage").getS());
-        }
-        if (item.containsKey("protect")) {
-            builder.protect(item.get("protect").getBOOL());
-        }
-        if (item.containsKey("allowZeroBytes")) {
-            builder.allowZeroBytes(item.get("allowZeroBytes").getBOOL());
-        }
-        if (item.containsKey("secondaryMetricsReporting")) {
-            builder.secondaryMetricsReporting(item.get("secondaryMetricsReporting").getBOOL());
-        }
-        if (item.containsKey("mutableTime")) {
-            builder.mutableTime(new DateTime(Long.parseLong(item.get("mutableTime").getN()), DateTimeZone.UTC));
+    @VisibleForTesting
+    Optional<ChannelConfig> mapItem(Map<String, AttributeValue> item) {
+        ChannelConfig cfg = null;
+        try {
+            ChannelConfig.ChannelConfigBuilder builder = ChannelConfig.builder()
+                    .creationDate(new Date(Long.parseLong(item.get("date").getN())))
+                    .name(item.get("key").getS())
+                    .displayName(item.get("displayName").getS());
+            if (item.get("ttlDays") != null) {
```

Comment:
It'd be nice to find a way to avoid all the if statments with a null object pattern or helper. Something like:
```
builder.ttlDays(item.getOrDefault("ttlDays"), AttributeValue.ALL_NULLS).getN());
builder.keepForever(item.getOrDefault("keepForever"), AttributeValue.ALL_NULLS).getBOOL());
builder.description(item.getOrDefault("ttlDays"), AttributeValue.ALL_NULLS).getS());
```

Reply:
Another way I've seen is to done is something like 
```
Optional.ofNullable(item.get("blah"))
    .map(AttributeValue::getWhatever)
    .ifPresent(builder::blah)
```